models/repo/repo.go

```diff
@@ -468,16 +468,16 @@ func (repo *Repository) ComposeDocumentMetas() map[string]string {
 // GetBaseRepo populates repo.BaseRepo for a fork repository and
 // returns an error on failure (NOTE: no error is returned for
 // non-fork repositories, and BaseRepo will be left untouched)
-func (repo *Repository) GetBaseRepo() (err error) {
-	return repo.getBaseRepo(db.DefaultContext)
+func (repo *Repository) GetBaseRepo(ctx context.Context) (err error) {
+	return repo.getBaseRepo(ctx)
 }
 
 func (repo *Repository) getBaseRepo(ctx context.Context) (err error) {
```

Comment:
```suggestion
func (repo *Repository) GetBaseRepo(ctx context.Context) (err error) {
```
Could merge these two.